src/librustc_metadata/cstore_impl.rs

```diff
@@ -47,6 +47,9 @@ macro_rules! provide {
                 $tcx: TyCtxt<$lt>,
                 def_id_arg: T,
             ) -> <ty::queries::$name<$lt> as QueryConfig<$lt>>::Value {
+                let _prof_timer =
+                    $tcx.prof.generic_activity("metadata_decode_entry");
```

Comment:
Might it make sense to associate the query name here? We have it statically I believe so we could do so without allocating a string (e.g., `concat!("metadata_decode_entry(", stringify!($name), ")")` or so)

Reply:
In the future yes. There's the plan to support recording event parameters (i.e. query keys and function arguments). That will need a bit of addition infrastructure in order to be efficient. And I suspect that it will be slow even then, so it will be behind an extra filter flag.